ginga/misc/plugins/SaveImage.py

```diff
@@ -0,0 +1,477 @@
+                    subs = ind + subs.lstrip()
+                hdu.header.add_history(subs)
+
+    def _write_header(self, image, hdu):
+        """Write header from image object to given HDU."""
+        hduhdr = hdu.header
+
+        # Ginga image header object for the given extension only.
+        # Cannot use get_header() because that might also return PRI hdr.
+        ghdr = image.metadata['header']
+
+        for key in ghdr:
+            # Need this to avoid duplication because COMMENT is a weird field
+            if key.upper() == 'COMMENT':
+                continue
+
+            bnch = ghdr.get_card(key)
+
+            # Insert new keyword
+            if key not in hduhdr:
+                hduhdr[key] = (bnch.value, bnch.comment)
+
+            # Update existing keyword
+            elif hduhdr[key] != bnch.value:
+                hduhdr[key] = bnch.value
+
+    def _write_mosaic(self, key, outfile):
+        """Write out mosaic data (or any new data generated within Ginga)
+        to single-extension FITS.
+
+        """
+        maxsize = self.settings.get('max_mosaic_size', 1e8)  # Default 10k x 10k
+        image = self.chinfo.datasrc[key]
+
+        # Prevent writing very large mosaic
+        if (image.width * image.height) > maxsize:
+            s = 'Mosaic too large to be written {0}'.format(image.shape)
+            self.w.status.set_text(s)
+            self.logger.error(s)
+            return
+
+        # Insert mosaic data and header into output HDU
+        hdu = fits.PrimaryHDU(image.get_data())
+        self._write_header(image, hdu)
+
+        # Write history to PRIMARY
+        self._write_history(key, hdu)
+
+        # Write to file
+        hdu.writeto(outfile, clobber=True)
+
+    def _write_mef(self, key, extlist, outfile):
+        """Write out regular multi-extension FITS data."""
+        with fits.open(outfile, mode='update') as pf:
+            # Process each modified data extension
+            for idx in extlist:
+                k = '{0}[{1}]'.format(key, self._format_extname(idx))
+                image = self.chinfo.datasrc[k]
+
+                # Insert data and header into output HDU
+                pf[idx].data = image.get_data()
+                self._write_header(image, pf[idx])
+
+            # Write history to PRIMARY
+            self._write_history(key, pf['PRIMARY'])
+
+    def toggle_save_cb(self, w, res_dict):
+        """Only enable saving if something is selected."""
+        if len(res_dict) > 0:
+            self.w.save.set_enabled(True)
+        else:
+            self.w.save.set_enabled(False)
+        return
+
+    def save_images(self):
+        """Save selected images."""
+        res_dict = self.treeview.get_selected()
+        clobber = self.settings.get('clobber', False)
+        self.treeview.clear_selection()  # Automatically disables Save button
+
+        # If user chooses fitsio over astropy, cannot continue.
+        if (io_fits.fitsLoaderClass.__name__ !=
```

Comment:
Can't then load the image via `fitsio`?  It looks like you are importing `astropy.io.fits` in this plugin so that is a dependency to use the plugin (fine), but it shouldn't really matter how the image came in, right?